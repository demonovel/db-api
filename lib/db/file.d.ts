/**
 * Created by user on 2020/5/18.
 */
import { IFileRecordRow, IFileRecordData, IFetchRecordOptionsBase } from '../types';
import Bluebird from 'bluebird';
export interface IFileRecordOptions extends IFetchRecordOptionsBase {
    siteID: string;
    novelID: string | number;
}
export declare function getFileRecord<T extends IFileRecordData = IFileRecordData>(options: IFileRecordOptions): Bluebird<IFileRecordRow<T>>;
export declare function putFileRecord<T extends IFileRecordData = IFileRecordData>(options: IFileRecordOptions & {
    data: T;
}): Bluebird<IFileRecordRow<T>>;
