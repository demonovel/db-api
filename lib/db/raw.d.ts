import { IRawRecordRow, IFetchRecordOptionsBase } from '../types';
import Bluebird from 'bluebird';
export interface IRawRecordOptions extends IFetchRecordOptionsBase {
    rootKey: string;
    dataKey: string | number;
}
export declare function getRawRecord<T>(options: IRawRecordOptions): Bluebird<IRawRecordRow<T>>;
export declare function putRawRecord<T extends any = unknown>(options: IRawRecordOptions & {
    data: T;
}): Bluebird<IRawRecordRow<T>>;
