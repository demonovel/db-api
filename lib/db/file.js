"use strict";
/**
 * Created by user on 2020/5/18.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.putFileRecord = exports.getFileRecord = void 0;
const util_1 = require("../util");
const fetch_1 = __importDefault(require("../fetch"));
function getFileRecord(options) {
    var _a, _b;
    let url = util_1.newFileURL(options.siteID, String(options.novelID), options.server);
    return fetch_1.default(url.href, {
        ...options.fetchOptions,
        method: 'GET',
        timeout: (_b = (_a = options.fetchOptions) === null || _a === void 0 ? void 0 : _a.timeout) !== null && _b !== void 0 ? _b : 60 * 1000,
    });
}
exports.getFileRecord = getFileRecord;
function putFileRecord(options) {
    var _a, _b;
    let url = util_1.newFileURL(options.siteID, String(options.novelID), options.server);
    return fetch_1.default(url.href, {
        ...options.fetchOptions,
        body: JSON.stringify(options.data),
        timeout: (_b = (_a = options.fetchOptions) === null || _a === void 0 ? void 0 : _a.timeout) !== null && _b !== void 0 ? _b : 60 * 1000,
    });
}
exports.putFileRecord = putFileRecord;
//# sourceMappingURL=file.js.map