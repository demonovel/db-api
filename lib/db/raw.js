"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.putRawRecord = exports.getRawRecord = void 0;
const tslib_1 = require("tslib");
const util_1 = require("../util");
const fetch_1 = (0, tslib_1.__importDefault)(require("../fetch"));
function getRawRecord(options) {
    var _a, _b;
    let url = (0, util_1.newFileURL)(options.rootKey, String(options.dataKey), options.server);
    return (0, fetch_1.default)(url.href, {
        ...options.fetchOptions,
        method: 'GET',
        timeout: (_b = (_a = options.fetchOptions) === null || _a === void 0 ? void 0 : _a.timeout) !== null && _b !== void 0 ? _b : 60 * 1000,
    });
}
exports.getRawRecord = getRawRecord;
function putRawRecord(options) {
    var _a, _b, _c, _d, _e;
    let url = (0, util_1.newFileURL)(options.rootKey, String(options.dataKey), options.server);
    return (0, fetch_1.default)(url.href, {
        ...options.fetchOptions,
        method: ((_c = (_b = (_a = options.fetchOptions) === null || _a === void 0 ? void 0 : _a.method) === null || _b === void 0 ? void 0 : _b.toUpperCase) === null || _c === void 0 ? void 0 : _c.call(_b)) === 'PUT' ? 'PUT' : 'POST',
        body: JSON.stringify(options.data),
        timeout: (_e = (_d = options.fetchOptions) === null || _d === void 0 ? void 0 : _d.timeout) !== null && _e !== void 0 ? _e : 60 * 1000,
    });
}
exports.putRawRecord = putRawRecord;
//# sourceMappingURL=raw.js.map