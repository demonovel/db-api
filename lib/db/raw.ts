import { IFileRecordData, IFileRecordRow, IRawRecordRow, IFetchRecordOptionsBase, EnumApiType } from '../types';
import { newFileURL } from '../util';
import fetch from '../fetch';
import { RequestInit } from 'node-fetch';
import Bluebird from 'bluebird';

export interface IRawRecordOptions extends IFetchRecordOptionsBase
{
	rootKey: string;
	dataKey: string | number;
}

export function getRawRecord<T>(options: IRawRecordOptions, type: EnumApiType = EnumApiType.raw)
{
	let url = newFileURL(options.rootKey, String(options.dataKey), options.server, type);

	return fetch<IRawRecordRow<T>>(url.href, {
		...options.fetchOptions,
		method: 'GET',
		timeout: options.fetchOptions?.timeout || 60 * 1000,
	})
}

export function putRawRecord<T extends any = unknown>(options: IRawRecordOptions & {
	data: T
}, type: EnumApiType = EnumApiType.raw)
{
	let url = newFileURL(options.rootKey, String(options.dataKey), options.server, type);

	return fetch<IRawRecordRow<T>>(url.href, {
		...options.fetchOptions,
		method: options.fetchOptions?.method?.toUpperCase?.() === 'PUT' ? 'PUT' : 'POST',
		body: JSON.stringify(options.data),
		timeout: options.fetchOptions?.timeout || 60 * 1000,
	})
}
