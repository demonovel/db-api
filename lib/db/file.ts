/**
 * Created by user on 2020/5/18.
 */

import { IFileRecordRow, IFileRecordData, IFetchRecordOptionsBase, EnumApiType } from '../types';
import { RequestInit } from 'node-fetch';
import { newFileURL } from '../util';
import fetch from '../fetch';
import Bluebird from 'bluebird';
import { getRawRecord, putRawRecord } from './raw';

export interface IFileRecordOptions extends IFetchRecordOptionsBase
{
	siteID: string;
	novelID: string | number;
}

export function getFileRecord<T extends IFileRecordData = IFileRecordData>(options: IFileRecordOptions): Bluebird<IFileRecordRow<T>>
{
	return getRawRecord<T>({
		...options,
		rootKey: options.siteID,
		dataKey: options.novelID,
	}, EnumApiType.file)
}

export function putFileRecord<T extends IFileRecordData = IFileRecordData>(options: IFileRecordOptions & {
	data: T
}): Bluebird<IFileRecordRow<T>>
{
	return putRawRecord<T>({
		...options,
		rootKey: options.siteID,
		dataKey: options.novelID,
	}, EnumApiType.file)
}

