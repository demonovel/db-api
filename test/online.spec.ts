import { getFileRecord } from '../lib/db/file';
import { RequestInit } from 'node-fetch';

test(`getFileRecord:error`, async (done) =>
{

	let actual = await getFileRecord({
			siteID: String(Date.now()),
			novelID: Date.now(),
		})
		.catch((r: any) => r.json())
	;

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot({
		error: true,
		timestamp: expect.any(Number),
		message: expect.any(String),
	});

	done();

});

test(`getFileRecord`, async (done) =>
{

	let actual = await getFileRecord({
			siteID: `wenku8`,
			novelID: 1695,
		})
	;

	//console.dir(actual)

	//expect(actual).toStrictEqual(expected);
	//expect(actual).toBeInstanceOf(Date);
	expect(actual).toMatchSnapshot({
		error: false,
		timestamp: expect.any(Number),
		data: {
			exists: true,
			filename: expect.any(String),
			href: expect.any(String),
			timestamp: expect.any(Number)
		}
	});

	done();

});

