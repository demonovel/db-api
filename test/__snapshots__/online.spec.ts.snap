// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFileRecord 1`] = `
{
  "data": {
    "exists": true,
    "filename": Any<String>,
    "href": Any<String>,
    "timestamp": Any<Number>,
  },
  "error": false,
  "timestamp": Any<Number>,
}
`;

exports[`getFileRecord:error 1`] = `
{
  "error": true,
  "message": Any<String>,
  "timestamp": Any<Number>,
}
`;
